### ISSUE TITLE
implement, then specify, extensions for private messaging

### ISSUE THREAD

**Issue by slingamn** [oragono/oragono#367]
Tracking ticket for:

1. Implementing, then specifying, a `deliver-to` extension
1. Implementing, then specifying, a way to do discovery from accounts to nicknames

**Comment by DanielOaks** [comment]
mentioned in-channel: make sure any caps we create are vendor-prefixed, e.g. `oragono.io/deliver-to`

**Comment by slingamn** [comment]
Significantly better ideas are being floated here: https://github.com/ircv3/ircv3-ideas/issues/34

although they don't solve the issue we discussed in #oragono of allowing people to "hide" by changing their nicknames.

**Comment by slingamn** [comment]
I'm no longer enthusiastic about this. I think the only way forward is to fully converge the nickname and account concepts. We're most of the way there with always-on clients.

**Comment by DanielOaks** [comment]
Yeah I agree

**Comment by slingamn** [comment]
Discussed: no need to close off the possibility of an account-first system, let's leave this open for further comment.

**Comment by DanielOaks** [comment]
rough ideas from me, everyone else can take or totally ignore these:

just flat-out privmsging `@accountname` and that being the target of the message on receiving clients that have negotiated a certain capability (if the cap isn't negotiated they'll just see them as regular self-messages).

that ole' `buddy-list` capability idea.

tl;dr keep all the traditional irc workings underneath whoo yay but legitimately just build a more consistent account-first system on top. can/should we remake msn?? who knows but we (I) can try!!!
